modules/sara_r4/cellular_r4_api.c

```diff
@@ -907,17 +907,20 @@ CellularError_t Cellular_SocketConnect( CellularHandle_t cellularHandle,
                            socketHandle->remoteSocketAddress.ipAddress.ipAddress,
                            socketHandle->remoteSocketAddress.port );
 
+        /* Set the socket state to connecting state. If cellular modem returns error,
+         * revert the state to allocated state. */
+        socketHandle->socketState = SOCKETSTATE_CONNECTING;
+
         pktStatus = _Cellular_TimeoutAtcmdRequestWithCallback( pContext, atReqSocketConnect,
                                                                SOCKET_CONNECT_PACKET_REQ_TIMEOUT_MS );
 
         if( pktStatus != CELLULAR_PKT_STATUS_OK )
         {
             LogError( ( "Cellular_SocketConnect: Socket connect failed, cmdBuf:%s, PktRet: %d", cmdBuf, pktStatus ) );
             cellularStatus = _Cellular_TranslatePktStatus( pktStatus );
-        }
-        else
-        {
-            socketHandle->socketState = SOCKETSTATE_CONNECTING;
+
+            /* Revert the state to allocated state. */
+            socketHandle->socketState = SOCKETSTATE_ALLOCATED;
```

Comment:
Is there a unit test that covers this change?

Reply:
We do have unit test for cellular common, which is 3GPP AT commands.
This change is for partner supported cellular module.
Unit test for cellular project doesn't cover module specific implementation.